### PR TITLE
Integrate golangci-lint and enable scope linting

### LINKED ISSUE #342: Add lint tool/check to detect pointers to loop variables
reference: https://github.com/open-telemetry/opentelemetry-service/pull/341

We should have a lint tool or check to warn against such cases.

### PR DESCRIPTION
**Pull request by ferhatelmas**
* integrate golangci-lint to easily extend linting capabilities later
  - that's why separate fmt, vet, etc. targets are dropped and all are run through umbrella lint target, golangci-lint.
* additionally enabled scopelint and fixed one issue
  - it's disabled for test files because it gives false positive - https://github.com/kyoh86/scopelint/issues/4.
* misspell on go files are also run by golangci-lint, that's why `ALL_SRC_AND_DOC` is simplified to `ALL_DOC`.

fixes #342

### CODE REVIEW

**Comment by owais** [comment]
Thanks. Did you evaluate other tools like gometalinter or the newer revive? (https://github.com/mgechev/revive)

Not saying we shouldn't use golangci-lint but would be nice to evaluate all available tools before we pick one.

**Comment by ferhatelmas** [comment]
gometalinter is dead and they suggest using golangci-lint in their readme. https://github.com/alecthomas/gometalinter#go-meta-linter

revive is a linter like gofumpt or golint which can be run through golangci-lint so I would say it's a different tool. (right now it should be run separately if needed since its including PR not merged yet https://github.com/golangci/golangci-lint/issues/238)

In the current state, golangci-lint seems the best to run other linters.

**Comment by ferhatelmas** [comment]
Btw, go sdk is already using it. https://github.com/open-telemetry/opentelemetry-go/blob/master/.golangci.yml

File: Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@ ALL_SRC := $(shell find . -name '*.go' \
                                 -type f | sort)
 
 # All source code and documents. Used in spell check.
-ALL_SRC_AND_DOC := $(shell find . \( -name "*.md" -o -name "*.go" -o -name "*.yaml" \) \
+ALL_DOC := $(shell find . \( -name "*.md" -o -name "*.yaml" \) \
```
Comment (tigrannajaryan) [inline Makefile]: Why is *.go removed?

File: Makefile [hunk 2]
```diff
@@ -56,7 +52,7 @@ benchmark:
 	$(GOTEST) -bench=. -run=notests $(ALL_PKGS)
 
 .PHONY: travis-ci
-travis-ci: fmt impi vet lint goimports misspell staticcheck test-with-cover otelcol
+travis-ci: impi lint misspell test-with-cover otelcol
```
Comment (tigrannajaryan) [inline Makefile]: Why are fmt,vet,goimports,staticcheck removed?

File: Makefile [hunk 3]
```diff
@@ -80,55 +76,17 @@ addlicense:
...
-		echo "$$IMPORTSOUT\n"; \
-		exit 1; \
-	else \
-	    echo "Goimports finished successfully"; \
-	fi
-
 .PHONY: misspell
 misspell:
-	$(MISSPELL) $(ALL_SRC_AND_DOC)
+	$(MISSPELL) $(ALL_DOC)
```
Comment (owais) [inline Makefile]: I've found it helpful to run this check on code as well in the past. Often catches bad variable/string names. Plus we have docs in .go files as well.

File: .golangci.yml [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+    - misspell
+    - scopelint
+
+issues:
+  # List of regexps of issue texts to exclude, empty list by default.
+  # But independently from this option we use default exclude patterns,
+  # it can be disabled by `exclude-use-default: false`. To list all
+  # excluded by default patterns execute `golangci-lint run --help`
+  exclude:
+    - abcdef
```
Comment (owais) [inline .golangci.yml]: Is this just here as an example? Can we remove it and keep `exclude` as an empty list or even comment out?

File: Makefile (hunk 3 above)
Comment (owais) [reply Makefile]: Ah, I see it is run by the umbrella lint tool now. Nice.

File: Makefile (hunk 2 above)
Comment (ferhatelmas) [reply Makefile]: they are run by umbrella tool.

**Comment by owais** [comment]
LGTM. I could not find any output in the CI build related to this. I suppose it is completely silent unless there is an issue to report?

File: Makefile (hunk 1 above)
Comment (ferhatelmas) [reply Makefile]: same, run by umbrella tool.

File: .golangci.yml (hunk 4 above)
Comment (ferhatelmas) [reply .golangci.yml]: yes, I didn't reduce the config and kept the examples. I wanted to see if the general idea is fine. If so, I will trim all extras.

**Comment by ferhatelmas** [comment]
> I suppose it is completely silent unless there is an issue to report?

yes but I can enable debug mode to output some info about runs.

File: Makefile (hunk 2 above)
Comment (tigrannajaryan) [reply Makefile]: OK. Please modify PR description/commit message to explain that.

Comment (ferhatelmas) [reply Makefile]: Sure, updating.

**Comment by ferhatelmas** [comment]
@owais removed `@` to show the command but didn't added `-v` flag because it adds a lot of noise. If you wish, I can add it conditionally (if Travis=true)

File: .golangci.yml (hunk 4 above)
Comment (owais) [reply .golangci.yml]: I think examples are good but better off if commented out.

**Comment by ferhatelmas** [comment]
Anything is needed from me?

**Pull request merged**